shell/platform/fuchsia/flutter/flatland_connection.cc

```diff
@@ -37,6 +37,10 @@ FlatlandConnection::~FlatlandConnection() = default;
 
 // This method is called from the raster thread.
 void FlatlandConnection::Present() {
+  if (!threadsafe_state_.first_present_called_) {
```

Comment:
Flutter animator waits for a Vsync signal from Flatland before starting to render a frame, see Animator::RequestFrame(). Flatland returns Vsync signal to the client through OnNextFrameBegin(), but only after they send their first Present(). |first_present_called_| was used to fake the first vsync signal in AwaitVsync(). We then follow Present()-OnNextFrame() sequence for the render loop.

Flatland's OnGetLayout() returns values to Flutter on a separate protocol and platform thread. Ideally, the clients should wait for it before their first Present() so they know how big their content should be. Flutter does that. However, it can take a couple AwaitVsync calls to have the size. 

All Flatland clients start with 1 present credit, so we know the first Present goes through.